Neos.Fusion/Classes/Core/ObjectTreeParser/PredictiveParser.php

```diff
@@ -0,0 +1,527 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Fusion\Core\ObjectTreeParser;
+
+/*
+ * This file is part of the Neos.Fusion package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Fusion;
+
+use Neos\Fusion\Core\ObjectTreeParser\Ast\FusionFile;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\StatementList;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\AbstractStatement;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\IncludeStatement;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\ObjectStatement;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\Block;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\ObjectPath;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\AbstractPathSegment;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\MetaPathSegment;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\PrototypePathSegment;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\PathSegment;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\ValueAssignment;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\AbstractPathValue;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\FusionObjectValue;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\DslExpressionValue;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\EelExpressionValue;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\SimpleValue;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\CharValue;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\StringValue;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\ValueCopy;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\AssignedObjectPath;
+use Neos\Fusion\Core\ObjectTreeParser\Ast\ValueUnset;
+use Neos\Fusion\Core\ObjectTreeParser\ExceptionMessage\MessageCreator;
+use Neos\Fusion\Core\ObjectTreeParser\ExceptionMessage\MessageLinePart;
+use Neos\Fusion\Core\ObjectTreeParser\Exception\ParserException;
+use Neos\Fusion\Core\ObjectTreeParser\Exception\ParserUnexpectedCharException;
+
+/**
+ * The Fusion Parser Engine
+ */
+class PredictiveParser
```

Comment:
I somehow would expect this to be the called "ObjectTreeParser" since it is in the folder ObjectTreeParser.